chain/consensus/mir/state_manager.go

```diff
@@ -0,0 +1,231 @@
+package mir
+
+import (
+	"bytes"
+	"fmt"
+
+	"github.com/filecoin-project/mir/pkg/pb/checkpointpb"
+	"github.com/filecoin-project/mir/pkg/pb/commonpb"
+	"github.com/filecoin-project/mir/pkg/pb/requestpb"
```

Comment:
Why is mir using protobuf when the entire Filecoin protocol is cbor-based? Protobufs are going to be a pain to deal with in IPLD.

Reply:
Mir was @matejpavlovic's project before joining PL. The project used `protobuf` already and this is why we didn't include cbor in the integration. I am pro-cbor, but again, I'll defer this one to @matejpavlovic.